### ISSUE TITLE
False positive for discarded_notification_center_observer rule (reopen #3224)

### ISSUE THREAD

**Issue by Jumhyn** [realm/SwiftLint#3498]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The following code triggers the `discarded_notification_center_observer` rule:

```swift
self.observers.append(
    NotificationCenter.default.addObserver(
        forName: UIApplication.willEnterForegroundNotification,
        object: nil,
        queue: nil
    ) {
        print($0)
    }
)
```

It looks like a similar issue was fixed in #2685, but it didn't cover the case where the call begins on the next line.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
% swiftlint lint test.swift
Linting Swift files at paths test.swift
Linting 'test.swift' (1/1)
/path/to/test.swift:2:5: warning: Discarded Notification Center Observer Violation: When registering for a notification using a block, the opaque observer that is returned should be stored so it can be removed later. (discarded_notification_center_observer)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version 0.39.1.
* CocoaPods installation.

**Comment by ahti** [comment]
Can confirm, I'm seeing the same false positive with this code:

```swift
self.notificationHandles.append(
    NotificationCenter.default.addObserver(forName: .SomeNotificationName, object: nil, queue: .main) { note in
        // ...
    }
)

```

**Comment by jpsim** [comment]
Unfortunately there's no easy way to fix this rule to avoid these false positives, so I'm changing this rule to opt-in: #3537

I'm closing this ticket as a duplicate of #1398, which was opened before this one.
